File: apprefurbished/src/main/java/com/androidcollider/easyfin/debts/pay/PayDebtMVP.java
package com.androidcollider.easyfin.debts.pay;

import android.os.Bundle;

import androidx.annotation.Nullable;

import com.androidcollider.easyfin.common.view_models.SpinAccountViewModel;

import java.util.List;

import io.reactivex.Flowable;

/**
 * @author Ihor Bilous
 */

interface PayDebtMVP {

    interface Model {

        Flowable<List<SpinAccountViewModel>> getAllAccounts();

        Flowable<Boolean> payFullDebt(int idAccount,
                                      double accountAmount,
                                      int idDebt);

        Flowable<Boolean> payPartOfDebt(int idAccount,
                                        double accountAmount,
                                        int idDebt,
                                        double debtAmount);

        Flowable<Boolean> takeMoreDebt(int idAccount,
                                       double accountAmount,
                                       int idDebt,
                                       double debtAmount,
                                       double debtAllAmount);

        String prepareStringToParse(String value);

        String formatAmount(double amount);
    }

    interface View {

        void showAmount(String amount);

        void showName(String name);

        void setupSpinner();

        void showAccount(int position);

        void showMessage(String message);

        void openNumericDialog();

        void notifyNotEnoughAccounts();

        void disableAmountField();

        void setAccounts(List<SpinAccountViewModel> accountList);

        void performLastActionsAfterSaveAndClose();

        String getAmount();

        SpinAccountViewModel getAccount();

        List<SpinAccountViewModel> getAccounts();
    }

    interface Presenter {

        void setView(@Nullable PayDebtMVP.View view);

        void setArguments(Bundle args);

        void loadAccounts();

        void save();
    }
}

File: apprefurbished/src/main/java/com/androidcollider/easyfin/debts/pay/PayDebtMVP.kt
package com.androidcollider.easyfin.debts.pay

import android.os.Bundle
import com.androidcollider.easyfin.common.view_models.SpinAccountViewModel
import io.reactivex.rxjava3.core.Single

/**
 * @author Ihor Bilous
 */
interface PayDebtMVP {
    interface Model {
        val allAccounts: Single<List<SpinAccountViewModel>>
        fun payFullDebt(
            idAccount: Int,
            accountAmount: Double,
            idDebt: Int
        ): Single<Boolean>

        fun payPartOfDebt(
            idAccount: Int,
            accountAmount: Double,
            idDebt: Int,
            debtAmount: Double
        ): Single<Boolean>

        fun takeMoreDebt(
            idAccount: Int,
            accountAmount: Double,
            idDebt: Int,
            debtAmount: Double,
            debtAllAmount: Double
        ): Single<Boolean>

        fun prepareStringToParse(value: String?): String
        fun formatAmount(amount: Double): String
    }

    interface View {
        fun showAmount(amount: String?)
        fun showName(name: String?)
        fun setupSpinner()
        fun showAccount(position: Int)
        fun showMessage(message: String?)
        fun openNumericDialog()
        fun notifyNotEnoughAccounts()
        fun disableAmountField()
        fun performLastActionsAfterSaveAndClose()
        val amount: String
        val account: SpinAccountViewModel
        var accounts: List<SpinAccountViewModel>
    }

    interface Presenter {
        fun setView(view: View?)
        fun setArguments(args: Bundle?)
        fun loadAccounts()
        fun save()
    }
}